observability-test/spanner.ts

```diff
@@ -1298,3 +1320,967 @@ describe('E2E traces with async/await', async () => {
+
+    // Finally check for the collective expected event names.
+    const expectedEventNames = [
+      'Requesting 25 sessions',
+      'Creating 25 sessions',
+      'Requested for 25 sessions returned 25',
+      'Starting stream',
+      'Acquiring session',
+      'Waiting for a session to become available',
+      'Acquired session',
+      'Using Session',
+    ];
+    assert.deepStrictEqual(
+      actualEventNames,
+      expectedEventNames,
+      `Unexpected events:\n\tGot:  ${actualEventNames}\n\tWant: ${expectedEventNames}`
+    );
+  }
+
+  it('database.run with bad syntax: async/await', async () => {
+    const instance = spanner.instance('instance');
+    const database = instance.database('database');
+
+    try {
+      const [rows] = await database.run(selectSql1p);
+    } catch (e) {
+      // This catch is meant to ensure that we
+      // can assert on the generated spans.
+    } finally {
+      provider.forceFlush();
+    }
+
+    assertRunBadSyntaxExpectations();
+  });
+
+  it('database.run with bad syntax: callback', done => {
+    const instance = spanner.instance('instance');
+    const database = instance.database('database');
+
+    database.run(selectSql1p, (err, rows) => {
+      assert.ok(err);
+      provider.forceFlush();
+      assertRunBadSyntaxExpectations();
+      done();
+    });
+  });
+
+  function assertDatabaseRunPlusAwaitTransactionForAlreadyExistentData() {
+    traceExporter.forceFlush();
+    const spans = traceExporter.getFinishedSpans();
+    spans.sort((spanA, spanB) => {
+      return spanA.startTime < spanB.startTime;
+    });
+
+    const actualSpanNames: string[] = [];
+    const actualEventNames: string[] = [];
+    spans.forEach(span => {
+      actualSpanNames.push(span.name);
+      span.events.forEach(event => {
+        actualEventNames.push(event.name);
+      });
+    });
+
+    const expectedSpanNames = [
+      'CloudSpanner.Database.batchCreateSessions',
+      'CloudSpanner.SessionPool.createSessions',
+      'CloudSpanner.Snapshot.run',
+      'CloudSpanner.Snapshot.begin',
+      'CloudSpanner.Snapshot.runStream',
+      'CloudSpanner.Snapshot.begin',
+      'CloudSpanner.Transaction.commit',
+      'CloudSpanner.Transaction.commit',
+      'CloudSpanner.Database.runTransactionAsync',
+    ];
+    assert.deepStrictEqual(
+      actualSpanNames,
+      expectedSpanNames,
+      `span names mismatch:\n\tGot:  ${actualSpanNames}\n\tWant: ${expectedSpanNames}`
+    );
+    const spanSnapshotRun = spans[2];
+    assert.strictEqual(spanSnapshotRun.name, 'CloudSpanner.Snapshot.run');
+    const wantSpanErr = '6 ALREADY_EXISTS: ' + messageBadInsertAlreadyExistent;
+    assert.deepStrictEqual(
+      spanSnapshotRun.status.code,
+      SpanStatusCode.ERROR,
+      'Unexpected status code'
+    );
+    assert.deepStrictEqual(
+      spanSnapshotRun.status.message,
+      wantSpanErr,
+      'Unexpexcted error message'
+    );
+
+    const databaseBatchCreateSessionsSpan = spans[0];
```

Comment:
@surbhigarg92 can't we name these variable as batchCreateSessionSpan instead of databaseBatchCreateSessionsSpan?

Reply:
will it be a good idea, or is it fine as is?